src/main/java/dev/esophose/playerparticles/particles/spawning/SpigotParticleSpawner.java

```diff
@@ -28,13 +28,13 @@ public void display(ParticleEffect particleEffect, double offsetX, double offset
     }
 
     @Override
-    public void display(ParticleEffect particleEffect, ParticleColor color, Location center, boolean isLongRange, Player owner) {
+    public void display(ParticleEffect particleEffect, ParticleColor color, Location center, boolean isLongRange, Player owner, float size) {
         if (!particleEffect.hasProperty(ParticleProperty.COLORABLE))
             throw new ParticleColorException("This particle effect is not colorable");
 
         if (particleEffect == ParticleEffect.DUST && NMSUtil.getVersionNumber() >= 13) { // DUST uses a special data object for spawning in 1.13+
             OrdinaryColor dustColor = (OrdinaryColor) color;
-            DustOptions dustOptions = new DustOptions(dustColor.toSpigot(), Setting.DUST_SIZE.getFloat());
+            DustOptions dustOptions = new DustOptions(dustColor.toSpigot(), size != 0 ? size : Setting.DUST_SIZE.getFloat());
```

Comment:
Use size > 0 here instead of size != 0. You missed adding this to the dust color transition handling.